### PR TITLE
Issue #15214: migrated section 5.2.8 type variable names to whole config testing

### PR DESCRIPTION
**Pull request by Zopsss**
#15214

### CODE REVIEW

File: src/it/resources-noncompilable/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputRecordTypeParameterName.java [hunk 1]
```diff
@@ -1,54 +1,52 @@
...
 import java.io.Serializable;
 import java.util.LinkedHashMap;
 import org.w3c.dom.Node;
 
-/* Config:
- * pattern = "(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"
- *
- */
+/** some javadoc. Config: pattern = "(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)" */
 public record InputRecordTypeParameterName<t>(Integer x, String str) {
```
Comment (Zopsss) [inline src/it/resources-noncompilable/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputRecordTypeParameterName.java:10]: record didn't had proper indentation, maybe I forgot to format this file in previous issue. so diff has become huge, reviewing it without whitespace would be easy and also moved extra records inside main record.

File: src/it/resources/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputClassTypeParameterName.java [hunk 2]
```diff
@@ -24,11 +24,12 @@ <T extends Runnable> foo getShadow() {
     return null;
   }
 
-  static class Junk<$foo> { // violation 'Class type name .* must match pattern'
+  static class Junk<$foo> { // violation 'Class type name '\$foo' must match pattern'
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputClassTypeParameterName.java:27]: escaped name

File: src/it/resources/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputInterfaceTypeParameterName.java [hunk 3]
```diff
@@ -41,41 +44,54 @@ <T> void getMoreFoo() {}
   }
 }
 
+// violation below 'Top-level class Boo has to reside in its own source file.'
 interface Boo<Input> { // violation 'Interface type name 'Input' must match pattern'
   Input boo();
 }
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputInterfaceTypeParameterName.java:50]: moving extra interfaces/classes inside main class would make the file not compile-able so had to add these violations

**Review by Zopsss — COMMENTED** [review]
reviewing.......

**Review by romani — APPROVED** [review]
ok to merge.

review PR without spaces

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**